cfme/tests/networks/nuage/test_nuage_inventory.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import pytest
+from netaddr import IPAddress
+
+from cfme.networks.provider.nuage import NuageProvider
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.provider([NuageProvider], scope="module")]
+
+
+def test_subnet_details_stats(appliance, provider, with_nuage_sandbox):
+    """
+    This test creates Nuage enterprise and its entities, including subnet.
+    Then it validates inventory of created subnet.
+
+    Steps:
+        * Deploy some entities outside of MIQ (directly in the provider)
+        * Create dictionary with stats that will be validated
+        * Find created subnet in database and validate its stats
+    """
+    sandbox = with_nuage_sandbox
+    cidr = '{address}/{netmask}'.format(
+        address=sandbox['subnet'].address,
+        netmask=IPAddress(sandbox['subnet'].netmask).netmask_bits()
+    )
+    subnet_stats = {
+        'name_value': sandbox['subnet'].name,
+        'type_value': 'ManageIQ/Providers/Nuage/Network Manager/Cloud Subnet/L3',
+        'cidr_value': cidr,
+        'gateway_value': sandbox['subnet'].gateway,
+        'network_protocol_value': sandbox['subnet'].ip_type.lower(),
+        'network_manager_value': provider.name,
+        'cloud_tenant_value': sandbox['enterprise'].name,
+        'network_router_value': sandbox['domain'].name,
+        'network_ports_num': len(sandbox['subnet'].vports),
+        'security_groups_num': sum(len(port.policy_groups) for port in sandbox['subnet'].vports),
+    }
+    provider.refresh_provider_relationships()
+    subnet, _ = wait_for(
+        lambda: appliance.collections.nuage_network_subnets.find_by_ems_ref(sandbox['subnet'].id)
```

Comment:
Yeah so as discussed offline I'd prefer if each iteration of the VMBD probing would log a line like

```
Looking for subnet with ems_ref %s in the VMBD...
```

That's because when user runs the test, she can have no idea what the test is currently doing unless we log it.

Reply:
I see you put logging into `find_by_ems_ref` method directly, but we actually only want to log those strings here because we're looping and waiting. The method itself shouldn't log IMAO - if we call it from elsewhere, logging wouldn't be desired.